### PR TITLE
[DDW-255] Integrate Shelley Haskell node and Shelley cardano-wallet

### PR DESCRIPTION
**Pull request by disassembler**
This PR adds support for cardano-wallet-shelley.

Which wallet added to bridge is now toggled by `useByronWallet` flag in envCfg. It also sets `kind` to `shelley` if `useByronWallet` is false. This will be simplified when the wallet has hard fork support and we combine shelley and byron wallets into the same binary.

What needs done in follow-ups at some point:

- [x] Integrate Daedalus and launcher to use the new binary
- [ ] Replace individual wallets with cross-era wallet when it's ready
- [ ] Update mac/windows installers to work with shelley environments (may make sense for the cross-era wallet to work first unless we need a public release of daedalus before the hardfork combinator is done) https://jira.iohk.io/browse/DDW-258
- [ ] Add support for `shelley_selfnode` https://jira.iohk.io/browse/DDW-257

## Todos

- [ ] Todo

## Screenshots

- Screenshot

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by nikolaglumac** [comment]
@disassembler I can't get Daedalus to sync up pass the 0 epoch, 114 slot stage. Any clue? Could this be related to the config? I can see the network hight rising...

**Comment by nikolaglumac** [comment]
@disassembler what do you mean by:
> Replace individual wallets with cross-era wallet when it's ready

Is it about Daedalus showing only Byron wallets before the hardfork and then showing both after the hardfork?

File: default.nix [hunk 1]
```diff
@@ -69,12 +69,17 @@ let
     sources = localLib.sources;
     bridgeTable = {
       jormungandr = self.callPackage ./nix/jormungandr-bridge.nix {};
-      cardano = self.callPackage ./nix/cardano-bridge.nix {};
+      cardano = self.callPackage ./nix/cardano-bridge.nix {
+        cardano-wallet = if self.launcherConfigs.launcherConfig.nodeConfig.kind == "byron"
+                         then self.cardano-wallet.cardano-wallet-byron
+                         else self.cardano-wallet.cardano-wallet-shelley;
+        cardanoWalletPkgs = self.cardano-wallet.pkgs;
+      };
```
Comment (cleverca22) [inline default.nix:77]: you could also potentially do `cardano-byron` and `cardano-shelley` in this table

File: installers/common/Types.hs [hunk 2]
```diff
@@ -120,7 +128,7 @@ packageFileName _os cluster ver backend _backendVer build = fromText name <.> ex
     name = T.intercalate "-" parts
     parts = ["daedalus", fromVer ver, lshowText cluster] ++ build'
     _backend' = case backend of
-                 Cardano _ -> "cardano-wallet"
+                 Cardano _ _ -> "cardano-wallet"
                  Jormungandr _ -> "jormungandr-wallet"
```
Comment (cleverca22) [inline installers/common/Types.hs:132]: is `_backend'` still used?

**Pull request merged**